varname.py

```diff
@@ -19,18 +19,51 @@
 class VarnameRetrievingError(Exception):
     """When failed to retrieve the varname"""
 
-def varname(caller: int = 1, raise_exc: bool = True) -> Optional[str]:
+def varname(caller: int = 1,
+            nvars: int = 1,
+            raise_exc: bool = True) -> Optional[Union[str, Tuple[str]]]:
     """Get the variable name that assigned by function/class calls
 
     Args:
         caller: The call stack index, indicating where this function
             is called relative to where the variable is finally retrieved
+        nvars: Number of variables to be expected on left-hand side (LHS).
+            if nvars > 1, a tuple of variable names will be returned.
+            Note that you also have to return the same number of elements in
+            your function:
+            For example:
+            >>> from varname import varname
+            >>> def func():
+            >>>   print(varname(nvars=2))
+            >>>   # return (1,) or (1,2,3) will raise an error
+            >>>   # not because of varname
+            >>>   # because you are trying to do: a, b = 1
+            >>>   return 1, 2
+            >>> a, b = func() # prints ('a', 'b')
+            >>> # (a, b) == (1, 2)
+
+            if nvars == 1, one can only do:
+            >>> var = func()
+            >>> (var, ) = func()
+            >>> # raise error, because you are trying to do (var, ) = 'var'
```

Comment:
`func()` doesn't return the varname, it returns `1, 2` so this explanation is confusing.

If `func()` did return varname, then of course this would be an unpacking error. But what happens with `(var, ) = func()` if it's meant to return a list of one element? Does varname() itself raise an error? The docs make it sound like it does. The code itself, if I'm reading correctly, will not raise an error, and varname() will return `var` either way. I think that's a good thing, for similar reasons as in my other comment, but it's not clear to me if that's what you intended.

Reply:
I was too cautious about what `func()` returns. Maybe I should just leave it there; it's just python's functionality. Trying to do this `(a, b) = 1` is of course an unpacking error. I should only talk about what `varname` returns, but not teach people how to write python functions.